include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3962,6 +4198,102 @@ DoFCellAccessor<DoFHandlerType, level_dof_access>::distribute_local_to_global(
 }
 
 
+
+template <typename DoFHandlerType, bool level_dof_access>
+inline bool
+DoFCellAccessor<DoFHandlerType, level_dof_access>::p_refine_flag_set() const
+{
+  Assert(this->used(), TriaAccessorExceptions::ExcCellNotUsed());
+  // cells flagged for refinement must be active
+  // (the @p set_p_refine_flag function checks this,
+  // but activity may change when refinement is
+  // executed and for some reason the refine
+  // flag is not cleared).
+  Assert(this->active() || !dealii::internal::DoFCellAccessorImplementation::
+                             Implementation::p_refine_flag_set(*this),
+         (typename CellAccessor<dim, spacedim>::ExcRefineCellNotActive()));
```

Comment:
The message only refers to the first part of the condition here. The second condition should be a separate assertion and message

Reply:
Are we able to translate the `OR` conditional into two separate ones? With DeMorgans laws I get `NOT( NOT(A) AND B)`, but either way I can't translate it into two independent ones. Or I don't get it.

Btw I copied the assertion from here:
https://github.com/dealii/dealii/blob/71f34a1297dde68c5169af4b72f16ddcc2b6149a/include/deal.II/grid/tria_accessor.templates.h#L3200

Reply:
Uh, doh! Yes. 

Would you be willing to adjust both error messages later on to describe the entirety of the condition, and not just one half?